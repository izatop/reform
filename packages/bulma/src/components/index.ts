export * from "./Breadcrumbs";
export * from "./Card";
export * from "./Drowdown";
export * from "./Level";
export * from "./Media";
export * from "./Menu";
export * from "./Message";
export * from "./Modal";
export * from "./Navbar";
export * from "./Pagination";
export * from "./Panel";
export * from "./Tabs";
