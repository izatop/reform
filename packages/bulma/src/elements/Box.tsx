import React from "react";
import {XProps} from "../interfaces";
import {MakeProps} from "../type";
import {ElementFactory} from "../utils";

const config = ElementFactory.create({component: "box"});
export const Box = config.factory<MakeProps, XProps<"div">>(({props, children}) => {
    return <div {...props}>{children}</div>;
});
