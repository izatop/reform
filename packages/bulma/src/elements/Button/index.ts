export * from "./Button";
export * from "./Buttons";
