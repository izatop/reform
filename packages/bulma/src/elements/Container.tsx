import React from "react";
import {XProps} from "../interfaces";
import {MakeBreakpoint} from "../type";
import {ElementFactory} from "../utils";

interface IContainer {
    fluid?: boolean;
}

const config = ElementFactory.create({component: "container"});
export const Container = config.factory<MakeBreakpoint<IContainer>, XProps<"div">>(({props, children}) => (
    <div {...props}>{children}</div>
));
