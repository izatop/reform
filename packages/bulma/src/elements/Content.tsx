import React from "react";
import {MakeProps, XProps} from "../interfaces";
import {ConfigFactory} from "../utils";

const config = ConfigFactory.create({component: "content"});
export const Content = config.factory<MakeProps, XProps<"div">>(({props, children}) => (
    <div {...props}>{children}</div>));
