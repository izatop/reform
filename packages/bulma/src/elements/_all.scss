@charset "utf-8";

@import "Button/all";
@import "Icon/all";
@import "Figure/all";
@import "Notification/all";
@import "Progress/all";
@import "Table/all";
@import "Tag/all";
@import "Title/all";
@import "Box.scss";
@import "Container.scss";
@import "Content.scss";
@import "Delete.scss";
