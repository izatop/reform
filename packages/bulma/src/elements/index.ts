export * from "./Button";
export * from "./Icon";
export * from "./Figure";
export * from "./Notification";
export * from "./Progress";
export * from "./Table";
export * from "./Tag";
export * from "./Title";
export * from "./Box";
export * from "./Container";
export * from "./Content";
export * from "./Delete";
export * from "./Node";
