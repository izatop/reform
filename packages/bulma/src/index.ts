export * from "./columns";
export * from "./components";
export * from "./elements";
export * from "./layout";
export * from "./forms";
export * from "./utils";
export * from "./interfaces";
export * from "./options";
export * from "./props";
