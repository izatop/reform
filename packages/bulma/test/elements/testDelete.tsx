import * as React from "react";
import * as renderer from "react-test-renderer";
import {Delete} from "../../src/elements";

test("Delete", () => {
    const div = renderer.create(<Delete size={"small"}/>);
    expect(div.toJSON()).toMatchSnapshot();
});
