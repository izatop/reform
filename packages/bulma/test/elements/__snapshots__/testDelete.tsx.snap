// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete 1`] = `
<a
  className="delete is-small"
/>
`;
